src/client/index.ts

```diff
@@ -1,6 +1,17 @@
+import { LDClient, LDFlagSet, LDUser } from 'launchdarkly-js-client-sdk';
+
 import useFlags from '../shared/useFlags';
 import useLDUser from '../shared/useLDUser';
 
 import LDBrowser from './ldBrowser';
 
+declare global {
+  interface Window {
+    ldClientBrowser: LDClient;
+    ssrFlags: LDFlagSet;
+    ldUser: LDUser;
+    clientSideID: string;
+  }
+}
+
```

Comment:
Fix for issues 2, 3, and 4. For some reason, `tsc` isn't able to access the `globals.d.ts` file in this project when client is pulled in.